File: ios-app/BayitPlusApp/Models/PodcastModels.swift
import Foundation

// MARK: - Podcasts

/// Response from GET /api/v1/podcasts
struct PodcastsResponse: Decodable, Sendable {
    let shows: [PodcastShow]
    let categories: [PodcastCategory]?
    let total: Int
    let page: Int
    let pages: Int
}

/// A podcast show
struct PodcastShow: Decodable, Sendable, Identifiable {
    let id: String
    let title: String?
    let author: String?
    let cover: String?
    let category: String?
    let cultureId: String?
    let episodeCount: Int?
    let latestEpisode: String?
    let availableLanguages: [String]?
}

/// A podcast category
struct PodcastCategory: Decodable, Sendable, Identifiable {
    let id: String
    let name: String
}

/// Response from GET /api/v1/podcasts/{show_id}
struct PodcastDetail: Decodable, Sendable, Identifiable {
    let id: String
    let title: String?
    let description: String?
    let author: String?
    let cover: String?
    let category: String?
    let website: String?
    let episodeCount: Int?
    let episodes: [PodcastEpisodeItem]?
    let latestEpisode: PodcastLatestEpisode?
}

/// Latest episode reference
struct PodcastLatestEpisode: Decodable, Sendable {
    let audioUrl: String?
}

/// A podcast episode
struct PodcastEpisodeItem: Decodable, Sendable, Identifiable {
    let id: String
    let title: String?
    let description: String?
    let audioUrl: String?
    let duration: String?
    let episodeNumber: Int?
    let seasonNumber: Int?
    let publishedAt: String?
    let thumbnail: String?
}

/// Response from GET /api/v1/podcasts/{show_id}/episodes
struct PodcastEpisodesResponse: Decodable, Sendable {
    let episodes: [PodcastEpisodeItem]
    let total: Int
    let page: Int
    let pages: Int
}

/// Response from GET /api/v1/podcasts/categories
struct PodcastCategoriesResponse: Decodable, Sendable {
    let categories: [PodcastCategory]
    let total: Int
}

/// Response from POST /api/v1/podcasts/{show_id}/sync
struct PodcastSyncResponse: Decodable, Sendable {
    let status: String
    let message: String?
    let episodesAdded: Int?
}

/// Response from POST /api/v1/podcasts/refresh
struct PodcastRefreshResponse: Decodable, Sendable {
    let status: String
    let message: String?
}

File: android-app/core/core-model/src/main/java/tv/bayit/plus/core/model/PodcastModels.kt
package tv.bayit.plus.core.model

import kotlinx.serialization.Serializable

/** Response from GET /api/v1/podcasts */
@Serializable
data class PodcastsResponse(
    val shows: List<PodcastShow> = emptyList(),
    val categories: List<PodcastCategory>? = null,
    val total: Int,
    val page: Int,
    val pages: Int,
)

/** A podcast show. */
@Serializable
data class PodcastShow(
    val id: String,
    val title: String? = null,
    val author: String? = null,
    val cover: String? = null,
    val category: String? = null,
    val cultureId: String? = null,
    val episodeCount: Int? = null,
    val latestEpisode: String? = null,
    val availableLanguages: List<String>? = null,
    val isSubscribed: Boolean? = null,
    val isUserAdded: Boolean? = null,
)

/** A podcast category. */
@Serializable
data class PodcastCategory(
    val id: String,
    val name: String,
)

/** Response from GET /api/v1/podcasts/{show_id} */
@Serializable
data class PodcastDetail(
    val id: String,
    val title: String? = null,
    val description: String? = null,
    val author: String? = null,
    val cover: String? = null,
    val category: String? = null,
    val website: String? = null,
    val episodeCount: Int? = null,
    val episodes: List<PodcastEpisodeItem>? = null,
    val latestEpisode: PodcastLatestEpisode? = null,
)

/** Latest episode reference. */
@Serializable
data class PodcastLatestEpisode(
    val audioUrl: String? = null,
)

/** A podcast episode. */
@Serializable
data class PodcastEpisodeItem(
    val id: String,
    val title: String? = null,
    val description: String? = null,
    val audioUrl: String? = null,
    val duration: String? = null,
    val episodeNumber: Int? = null,
    val seasonNumber: Int? = null,
    val publishedAt: String? = null,
    val thumbnail: String? = null,
)

/** Response from GET /api/v1/podcasts/{show_id}/episodes */
@Serializable
data class PodcastEpisodesResponse(
    val episodes: List<PodcastEpisodeItem> = emptyList(),
    val total: Int,
    val page: Int,
    val pages: Int,
)

/** Response from GET /api/v1/podcasts/categories */
@Serializable
data class PodcastCategoriesResponse(
    val categories: List<PodcastCategory> = emptyList(),
    val total: Int,
)

/** Response from POST /api/v1/podcasts/{show_id}/sync */
@Serializable
data class PodcastSyncResponse(
    val status: String,
    val message: String? = null,
    val episodesAdded: Int? = null,
)

/** Response from POST /api/v1/podcasts/refresh */
@Serializable
data class PodcastRefreshResponse(
    val status: String,
    val message: String? = null,
)
